smack-core/src/main/java/org/jivesoftware/smack/c2s/StreamOpenAndCloseFactory.java

```diff
@@ -0,0 +1,26 @@
+/**
+ *
+ * Copyright 2020 Aditya Borikar.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smack.c2s;
+
+import org.jivesoftware.smack.packet.StreamClose;
+import org.jivesoftware.smack.packet.StreamOpen;
+
+public interface StreamOpenAndCloseFactory {
+    StreamOpen createStreamOpen(CharSequence to, CharSequence from, String id, String lang);
```

Comment:
I just thought some more about it. Could you create `AbstractStreamOpen` and subclass `StreamOpen`, and then in your websocket branch `WebsocketOpen` from it. Same goes probably for `StreamClose` → `AbstractStreamClose`.

Reply:
And bonus points if you do the AbstractStreamOpen/StreamOpen splits in an extra commit. :)

Reply:
Made changes accordingly in an extra commit :smiley: